app/connectors/Sub09Connector.scala

```diff
@@ -21,81 +21,98 @@ import models.responses.{MdgSub09CompanyInformationResponse, MdgSub09Response, M
 import models.{CompanyInformation, NotificationEmail, XiEoriAddressInformation, XiEoriInformation}
 import play.api.{Logger, LoggerLike}
 import services.MetricsReporterService
-import uk.gov.hmrc.http.HttpReads.Implicits._
-import uk.gov.hmrc.http.{HeaderCarrier, HttpClient, StringContextOps}
+import uk.gov.hmrc.http.HttpReads.Implicits.*
+import uk.gov.hmrc.http.client.HttpClientV2
+import uk.gov.hmrc.http.{HeaderCarrier, StringContextOps}
 import utils.DateTimeUtils.rfc1123DateTimeFormatter
-import utils.Utils.emptyString
+import utils.Utils.{emptyString, getUri}
 
 import java.time.LocalDateTime
 import java.util.UUID
 import javax.inject.Inject
 import scala.concurrent.{ExecutionContext, Future}
 import scala.util.Random
 
-
 class Sub09Connector @Inject()(appConfig: AppConfig,
-                               http: HttpClient,
+                               http: HttpClientV2,
                                metricsReporter: MetricsReporterService)(implicit executionContext: ExecutionContext) {
 
   val log: LoggerLike = Logger(this.getClass)
-  private val acknowledgementRefLength = 32
+
   private val metricsResourceName = "mdg.get.company-information"
   private val defaultConsent = "0"
+  private val endPoint = appConfig.sub09GetSubscriptionsEndpoint
+
+  private def localDate: String = LocalDateTime.now().format(rfc1123DateTimeFormatter)
 
   def getSubscriberInformation(eori: String): Future[Option[NotificationEmail]] = {
     implicit val hc: HeaderCarrier = HeaderCarrier()
 
     metricsReporter.withResponseTimeLogging(metricsResourceName) {
-      http.GET[MdgSub09Response](uri(eori), headers = createHeaders(localDate)).map {
-        case MdgSub09Response(Some(email), Some(timestamp)) => Some(NotificationEmail(email, timestamp, None))
-        case _ => None
-      }
+      http.get(getUri(eori, endPoint))
+        .setHeader(
+          "Authorization" -> appConfig.sub09BearerToken,
```

Comment:
Better to put the common Headers name in an object (named Headers) under config package like below and make use of these. Also write the tests

package config

object Headers {
  val AUTHORIZATION = "Authorization"
  val DATE = "Date"
  val X_CORRELATION_ID = "X-Correlation-ID"
  val X_FORWARDED_HOST = "X-Forwarded-Host"
  val ACCEPT = "Accept"
}